File: Python/Algorithms/Quicksort.py
# Easy, single parameter method call to get the method going.
def quickSort(arr):
    quickSortAux(arr, 0, len(arr))

# Actual quicksort algorithm
def quickSortAux(arr, low, high):
    if low < high:
        # Partition the array
        p = partition(arr, low, high)
        # Sort the left half
        quickSortAux(arr, low, p - 1)
        # Sort the right half
        quickSortAux(arr, p + 1, high)

# Partition 
def partition(arr, low, high):
    # Pick a pivot
    pivot = arr[high]
    i = low
    # If a number is less than the pivot, put it on the left
    # if more, put it on the right
    for j in range(low, high):
        if arr[j] < pivot:
            swap(arr, i, j)
            i += 1
    swap(arr, i, high)
    return i

# Auxiliary swap method
def swap(arr, a, b):
    tmp = arr[a]
    arr[a] = arr[b]
    arr[b] = tmp

File: PHP/Algorithms/Quicksort.php
<?php

class Quicksort {
    public static function sort(&$arr) {
        Quicksort::_quicksort($arr, 0, count($arr) - 1);
    }

    public static function _quicksort(&$arr, $low, $high) {
        if($low < $high) {
            $p = Quicksort::partition($arr, $low, $high);
            
            Quicksort::_quicksort($arr, $low, $p - 1);
            Quicksort::_quicksort($arr, $p + 1, $high);
        }
    }

    public static function partition(&$arr, $low, $high) {
        // Taking the high pivot method. 
        $pivot = $arr[$high];
        $i = $low;
        for($j = $low; $j < $high; $j++) {
            if($arr[$j] < $pivot) {
                Quicksort::swap($arr, $i, $j);
                $i++;
            }
        }
        Quicksort::swap($arr, $i, $high);
        return $i;
    }

    public static function swap(&$arr, $a, $b) {
        $tmp = $arr[$a];
        $arr[$a] = $arr[$b];
        $arr[$b] = $tmp;
    }
}

$arr = array(5,2,3,7,4);
Quicksort::sort($arr);
echo json_encode($arr);